src/Shared/TypeLoader.cs

```diff
@@ -41,6 +40,8 @@ internal class TypeLoader
         /// </summary>
         private static readonly Object s_reflectionOnlyloadInfoToTypeLock = new Object();
 
+        private static CoreClrAssemblyLoader s_coreClrAssemblyLoader;
```

Comment:
Shouldn't this be behind a feature flag of some sort?  How does this loader interact when we're run in the full framework?